Kinvey/KinveyTests/UserTests.swift

```diff
@@ -1828,5 +1828,177 @@ class UserTests: KinveyTestCase {
+                    client?.urlProtocol(self, didReceive: response, cacheStoragePolicy: .notAllowed)
+                    let json = [
+                        "_socialIdentity" : [
+                            "kinveyAuth": [
+                                "access_token" : "a10a3743028e2e92b97037825b50a2666608b874",
+                                "refresh_token" : "627b034f5ec409899252a8017cb710566dfd2620",
+                                "id" : "custom",
+                                "audience" : MockKinveyBackend.kid
+                            ]
+                        ],
+                        "username" : "3b788b0c-cb99-4692-b3ae-a6b10b3d76f2",
+                        "password" : "fa0f771f-6480-4f11-a11b-dc85cce52beb",
+                        "_kmd" : [
+                            "lmt" : "2016-09-01T01:48:01.177Z",
+                            "ect" : "2016-09-01T01:48:01.177Z",
+                            "authtoken" : "12ed2b41-a5a1-4f37-a640-3a9c62c3fefd.rUHKOlQuRb4pW4NjmCimJ64rd2BF3drXy1SjHtuVCoM="
+                        ],
+                        "_id" : "57c788d168d976c525ee4602",
+                        "_acl" : [
+                            "creator" : "57c788d168d976c525ee4602"
+                        ]
+                    ] as [String : Any]
+                    let data = try! JSONSerialization.data(withJSONObject: json, options: [])
+                    client?.urlProtocol(self, didLoad: data)
+                    client?.urlProtocolDidFinishLoading(self)
+                default:
+                    let response = HTTPURLResponse(url: request.url!, statusCode: 401, httpVersion: "HTTP/1.1", headerFields: ["Content-Type" : "application/json; charset=utf-8"])!
+                    client?.urlProtocol(self, didReceive: response, cacheStoragePolicy: .notAllowed)
+                    let json = [
+                        "error" : "InvalidCredentials",
+                        "description" : "Invalid credentials. Please retry your request with correct credentials",
+                        "debug" : "Error encountered authenticating against kinveyAuth: {\"error\":\"server_error\",\"error_description\":\"Access Token not found\"}"
+                    ]
+                    let data = try! JSONSerialization.data(withJSONObject: json, options: [])
+                    client?.urlProtocol(self, didLoad: data)
+                    client?.urlProtocolDidFinishLoading(self)
+                }
+                type(of: self).count += 1
+            }
+            
+            override func stopLoading() {
+            }
+        }
+        
+        setURLProtocol(MICLoginAutomatedAuthorizationGrantFlowURLProtocol.self)
+        defer {
+            setURLProtocol(nil)
+        }
+        
+        XCTAssertNil(client.activeUser)
+        
+        weak var expectationLogin = expectation(description: "Login")
+        
+        let redirectURI = URL(string: "micAuthGrantFlow://")!
+        User.login(
```

Comment:
@heyzooi Shouldn't this be `User.loginWithAuthorization` as described [here](http://devcenter.kinvey.com/ios/guides/mobile-identity-connect#automated-authorization-grant)?